nautobot/core/nautobot_app/__init__.py

```diff
@@ -0,0 +1,169 @@
+from django.apps import AppConfig
+from django.urls import reverse
+from operator import getitem
+from collections import OrderedDict
+
+from nautobot.extras.choices import CustomLinkButtonClassChoices
+from nautobot.extras.plugins.utils import import_object
+from nautobot.extras.registry import registry
+
+
+class NautobotConfig(AppConfig):
+    menu_tabs = "navigation.menu_tabs"
+
+    def ready(self):
+        if self.__module__.endswith(".apps"):
+            self.__module__ = self.__module__[:-5]
+        menu_items = import_object(f"{self.__module__}.{self.menu_tabs}")
+        if menu_items is not None:
+            register_menu_items(menu_items)
+
+
+def register_menu_items(class_list):
```

Comment:
Please add docstring, particularly including an example of how `class_list` is to be structured. (Would `tab_list` be a better name?)